sound/soc/sof/intel/hda.c

```diff
@@ -1176,8 +1176,12 @@ int hda_dsp_remove(struct snd_sof_dev *sdev)
 				SOF_HDA_INT_CTRL_EN | SOF_HDA_INT_GLOBAL_EN, 0);
 
 	/* disable cores */
-	if (chip)
-		hda_dsp_core_reset_power_down(sdev, chip->host_managed_cores_mask);
+	if (chip) {
+		if (chip->hw_ip_version > SOF_INTEL_CAVS_2_5)
+			mtl_dsp_power_down_dsp_subsystem(sdev);
```

Comment:
It's kind of ugly, we typically use abstractions rather than explicit check for version numbers.

Why not follow this example:
````c
bool hda_check_ipc_irq(struct snd_sof_dev *sdev)
{
	const struct sof_intel_dsp_desc *chip;

	chip = get_chip_info(sdev->pdata);
	if (chip && chip->check_ipc_irq)
		return chip->check_ipc_irq(sdev);

	return false;
}
`````

Reply:
ACK, coming up with the patches. As soon as testing is done, I will push this again